Update Configure Your Cluster page including Worker Node Pool, tags and View Cluster Settings (#2212)
astronomer/docs: astro/modify-cluster.md

description: Learn what you can configure on an existing Astro cluster.
---

Astro cluster runs your Astro Deployments in isolated namespaces on the Kubernetes cluster. Your organization might choose to have one or multiple clusters to meet certain networking, governance, or use case requirements. You can always start with one cluster and [create additional clusters](create-cluster.md) later. New clusters on Astro typically have default configurations that are suitable for standard use cases. You can request [Astronomer support](https://support.astronomer.io) to edit the configuration of an existing cluster. For example, you might need to:

> - Add a [worker type](modify-cluster.md#about-worker-node-pools), which creates a new [worker node pool](modify-cluster.md#about-worker-node-pools) in your cluster and allows your team to select that worker type in a Deployment

Reviewer: ```suggestion
- Add a [worker type](#manage-worker-type), which creates a new [worker node pool](#about-worker-node-pools) in your cluster and allows your team to select that worker type in a Deployment.
```
You only need to add the anchor for a topic if the topic is in the same doc you're writing in.

- Increase or decrease the maximum number of worker nodes for a given worker type that your cluster can scale-up to. 
- Create a VPC connection or a transit gateway connection between your Astro cluster and a target VPC 

Cluster modifications typically take only a few minutes to complete and don't require downtime. In these cases, the Cloud UI and Airflow UI continue to be available and your Airflow tasks are not interrupted.

If you don't have a cluster on Astro, see [Install Astro](https://docs.astronomer.io/astro/category/install-astro).

## View clusters

In the **Cloud UI**, go to your organization's home page by clicking on the Astronomer icon ![icon](/img/favicon.ico) on the top left-hand-side corner. Then, click the **Clusters** tab to view a list of the clusters that are available to your organization. Click a cluster to view it's Cloud provider, region, creation date and last update date. Refer the below table for additional cluster information you can access: 

> | Tab name | Information | Access | 

Reviewer: Not sure what the access tab is communicating here. I think we can just link out to the relevant docs in the "Description" section.

|----------|-------------|---------|
| Details  | - Name <br/> - VPC Subnet Range <br/> - DB Instance Type <br/> - Account ID/Project ID/Subscription ID/Tenant ID <br/> - External IPs | You can only view the details from Cloud UI |
| Worker Types | - Node Instance Type <br/> - Max Node Count| You can only view the details from Cloud UI | 
| Tags | All tags that exist on the cluster | You can only view the details from Cloud UI | 
| Workspace Authorization | List of Workspaces that can create Deployments on this cluster | You can restrict or authorize workspaces to a cluster. See [Authorize Workspaces to a cluster](modify-cluster.md#authorize-workspaces-to-a-cluster) |

## Manage worker types

> On Astro, _worker nodes_ are the nodes that are used to run Airflow workers, which are responsible for executing Airflow tasks in your Deployments. A _worker type_ is one of your cloud provider's available node instance types. This determines how much CPU and memory your workers have for running tasks. Workers are organized using [worker node pools](modify-cluster.md#about-worker-node-pools) grouping nodes of a specific instance type.

Reviewer: ```suggestion
On Astro, _worker nodes_ are the nodes that are used to run Airflow workers, which are responsible for executing Airflow tasks in your Deployments. A _worker type_ is one of your cloud provider's available node instance types. This determines how much CPU and memory your workers have for running tasks. Workers are organized using [worker node pools](modify-cluster.md#about-worker-node-pools) that group nodes of a specific instance type.
```

> To have a Deployment run Airflow tasks with a specific worker type, you can configure a Worker queue in your Deployment to use that worker type. If the worker type is not available to your cluster, you can raise a request with [Astronomer support](https://support.astronomer.io) with the name of the worker type to enable it for your cluster.

Reviewer: ```suggestion
To have a Deployment run Airflow tasks with a specific worker type, you can configure a Worker queue in your Deployment to use that worker type. If the worker type is not available to your cluster, you can contact [Astronomer support](https://support.astronomer.io) and ask to have a worker type enabled for your cluster.
```

> To confirm a modification was completed, you can [check your cluster settings](modify-cluster.md#view-clusters).

Reviewer: ```suggestion
After you make a change to your available worker types, you can [check your cluster settings](modify-cluster.md#view-clusters) to confirm that the change was applied.
```

To have a Deployment run Airflow tasks with a specific worker type, you can configure a Worker queue in your Deployment to use that worker type. If the worker type is not available to your cluster, you can contact [Astronomer support](https://support.astronomer.io) and ask to have a worker type enabled for your cluster.

After you make a change to your available worker types, you can [check your cluster settings](modify-cluster.md#view-clusters) to confirm that the change was applied.

### About worker node pools

A Kubernetes _node pool_ is group of nodes within a cluster that share the same configuration. On Astro, a _worker node pool_ is a Kubernetes node pool that's used to run Airflow workers, which are responsible for executing Airflow tasks in your Deployments. Each worker node pool has:

- A _worker type_, which is one of your cloud provider's available node instance types.

> - A _maximum node count_, which is the the maximum number of nodes that can run concurrently in the worker node pool across all Deployments.

Reviewer: ```suggestion
- A _maximum node count_, which is the maximum number of nodes that can run concurrently in the worker node pool across all Deployments.
```

> During the Astro cluster creation process, you provide Astronomer with a default worker type and a maximum node count to configure a default worker node pool. This node pool is used for default Worker queues for all of your Deployments. When you request Astronomer support to add a [new worker type](modify-cluster.md#configure-node-instance-type) to your cluster, new worker node pool is created on the cluster. You can then configure [worker queues](configure-worker-queues.md) that use the new worker type. 

Reviewer: ```suggestion
During the Astro cluster creation process, you provide Astronomer with a default worker type and a maximum node count to configure a default worker node pool. This node pool is used for the default worker queues for all of your Deployments. When you request Astronomer support to add a [new worker type](modify-cluster.md#configure-node-instance-type) to your cluster, a new worker node pool of that type is created on the cluster. You can then configure [worker queues](configure-worker-queues.md) that use the new worker type. 
```

Your default worker node pool always runs a minimum of one node, while additional worker node pools can scale to zero. Additional nodes spin up as required based on the auto-scaling logic of your Deployment Airflow executor and your worker queues.

A worker node pool can be shared across multiple Deployments, but each node runs only the work of a single Deployment. A worker node pool's maximum node count applies across all Deployments where the node pool is used. For example, consider two Deployments that share a worker node pool with a maximum node count of 30. If one Deployment is using all 30 of the available worker nodes, the other Deployment can't use any additional nodes and therefore can't run tasks that rely on that worker type.

In the following diagram, you can see the relationship between worker node pools, Deployments, and worker queues. The worker node pools can be assigned to multiple Deployments, but each node in a given worker pool is used in only one Deployment worker queue.

![Worker Node Pool and Worker Queues](/img/docs/worker-node-pool.png)

### Configure node instance type

> Each worker type on Astro is configured with a node instance type that is defined by your cloud provider. For example, `m5.2xlarge` on AWS, `Standard_D8_v5` on Azure, or `e2-standard-8` on GCP. Node instance types are comprised of varying combinations of CPU, memory, storage, and networking capacity. By choosing a node instance type for your worker, you can provide the appropriate balance of resources for your Airflow tasks.

Reviewer: ```suggestion
Each worker type on Astro is configured with a node instance type that is defined by your cloud provider. For example, `m5.2xlarge` on AWS, `Standard_D8_v5` on Azure, or `e2-standard-8` on GCP. Node instance types consist of varying combinations of CPU, memory, storage, and networking capacity. By choosing a node instance type for your worker, you can provide the appropriate balance of resources for your Airflow tasks.
```

### About worker node pools

:::tip

- Node Pool, a Kubernetes Concept, is used to refer to a group of nodes within a cluster that have the same configuration. 
- In Astro Cluster, Worker Node Pool is a node pool for the Airflow Workers for an Airflow Deployment. 
- Worker Type is a node instance type (based on your choice of cloud) chosen for running Airflow Workers.

> - [Worker Queue](https://docs.astronomer.io/astro/configure-worker-queues) in Astro utilizes the concept of Celery Queues, when using Celery Executor, to segregate different types of tasks to achieve effective resource utilization. Individual tasks in a DAG can be assigned to a particular worker queue. 

Author: I want to move this with additional info about K8s executor and then try to refer it here. Because "Worker Queue" is also being used in Kubernetes even though not like in celery executor. it is more of the type of nodes a deployment is using

Reviewer: I don't think we need to overstress this - isn't the Kubernetes term "Work queue"?

### About worker node pools

A Kubernetes _node pool_ is group of nodes within a cluster that share the same configuration. On Astro, a _worker node pool_ is a Kubernetes node pool that's used to run workers, which are responsible for executing Airflow tasks in your Deployments. Each worker node pool has:

- A _worker type_, which is one of your cloud provider's available instance types. This determines how much CPU and memory your workers have for running tasks.
- A _maximum node count_, which is the the maximum number of nodes that can run concurrently in the worker node pool across all Deployments that use the worker queue.

> :::

Reviewer 2: ```suggestion

```

Your PR looks great and this is an awesome explanation. Since Jake already did a deep dive review, I won't add another change request, just a comment.

### About worker node pools

:::tip

> - Node Pool, a Kubernetes Concept, is used to refer to a group of nodes within a cluster that have the same configuration. 
> - In Astro Cluster, Worker Node Pool is a node pool for the Airflow Workers for an Airflow Deployment. 
> - Worker Type is a node instance type (based on your choice of cloud) chosen for running Airflow Workers.
> - [Worker Queue](https://docs.astronomer.io/astro/configure-worker-queues) in Astro utilizes the concept of Celery Queues, when using Celery Executor, to segregate different types of tasks to achieve effective resource utilization. Individual tasks in a DAG can be assigned to a particular worker queue. 

Reviewer: ```suggestion
A Kubernetes _node pool_ is group of nodes within a cluster that share the same configuration. On Astro, a _worker node pool_ is a Kubernetes node pool that's used to run workers, which are responsible for executing Airflow tasks in your Deployments. Each worker node pool has:

- A _worker type_, which is one of your cloud provider's available instance types. This determines how much CPU and memory your workers have for running tasks.
- A _maximum node count_, which is the the maximum number of nodes that can run concurrently in the worker node pool across all Deployments that use the worker queue.
```

To request support for a different database instance type or to modify the database instance type after cluster creation, contact [Astronomer support](https://cloud.astronomer.io/support).

> ## Add Tags to your Astronomer Infrastructure (AWS only)

Reviewer: ```suggestion
## Add tags to your cluster (AWS only)
```

Author: @jwitz these tags not only get added to the cluster but to other infrastructure as well. this is the reason I wrote Astronomer Infrastructure. WDYT should be correct here?

> You can request Astronomer Support to add Tags to the Astronomer infrastructure in your Data Plane account. The tags are applied to the Astronomer provisioned infrastructure, for example, EKS Cluster, RDS Metadata database, S3 Logs Bucket, IAM Roles. You can raise a request with [Astronomer Support](https://cloud.astronomer.io/support) to add or remove tags at any time. 

Reviewer: ```suggestion
You can request Astronomer Support to add [tags](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-tags.html) to your AWS clusters. The tags are applied to all Astronomer-provisioned infrastructure, such as your EKS cluster, RDS metadata database, S3 Logs Bucket, and IAM Roles. Tags can help your team identify your Astro clusters and associate them with a particular purpose or owner within your cloud provider ecosystem. 

You can raise a request with [Astronomer Support](https://cloud.astronomer.io/support) to add or remove tags. To view your current cluster tags, see [View clusters](#view-clusters).
```

Author: same question here @jwitz . In the first line, you have suggested AWS clusters and in the second we are specifying Astronomer-provisioned infrastructure.
